src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -477,6 +478,146 @@ pass::EliminateSqueeze::EliminateSqueeze() {
     this->register_matcher(m, callback);
 }
 
+namespace {
+bool check_squeeze(const shared_ptr<Node>& node) {
+    auto squeeze = std::dynamic_pointer_cast<ov::opset9::Squeeze>(node);
+    if (squeeze) {
+        auto axis = std::dynamic_pointer_cast<ov::opset9::Constant>(squeeze->input_value(1).get_node_shared_ptr());
+        if (axis) {
+            auto axis_val = axis->cast_vector<int64_t>();
+            if (axis_val.size() == 1 && axis_val[0] == 1) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool check_reshape(const shared_ptr<Node>& node) {
+    auto reshape = std::dynamic_pointer_cast<ov::opset9::Reshape>(node);
+    if (reshape) {
+        auto shape_pattern =
+            std::dynamic_pointer_cast<ov::opset9::Constant>(reshape->input_value(1).get_node_shared_ptr());
+        if (shape_pattern) {
+            auto pattern_val = shape_pattern->cast_vector<int64_t>();
+            pattern_val.insert(pattern_val.begin() + 1, 1);
+            auto in_shape = reshape->input_value(0).get_partial_shape();
+            if (in_shape.compatible(pattern_val)) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+template <class T>
+std::shared_ptr<T> check_all_inputs(const std::shared_ptr<ov::opset9::Concat>& concat) {
+    shared_ptr<T> split;
+    const auto concat_in_values = concat->input_values();
+    int idx = 0;
```

Comment:
```suggestion
    size_t idx = 0;
```
because concat_in_values.size() returns `size_t`.

Reply:
done